src/controllers/developers/utils.js

```diff
@@ -0,0 +1,72 @@
+import { developerSchema } from '../../models/Developer';
+import { createDbConnection } from '../../services/database';
+import { isTest } from '../../config';
+
+const PROD_LIMIT = 2500;
+export const FALLBACK_API_KEY = 'fallback_api_key';
+
+const determineLimit = (apiLimit) => (isTest ? apiLimit || PROD_LIMIT : PROD_LIMIT);
+export const fetchRequestQuery = (req) => {
+  const { apiLimit } = req.query;
+  return apiLimit;
+};
+
+const isSameDate = (first, second) => (
+  first.getFullYear() === second.getFullYear()
+    && first.getMonth() === second.getMonth()
+    && first.getDate() === second.getDate()
+);
+
+/* Increments usage count and updates usage date */
+const handleDeveloperUsage = async (developer) => {
+  const updatedDeveloper = developer;
+  const isNewDay = !isSameDate(updatedDeveloper.usage.date, new Date());
+  updatedDeveloper.usage.date = Date.now();
+
+  if (isNewDay) {
+    updatedDeveloper.usage.count = 0;
+  } else {
+    updatedDeveloper.usage.count += 1;
+  }
+
+  return updatedDeveloper.save();
+};
+
+export const fetchAPIKey = (req) => {
+  const apiKey = req.headers['X-API-Key'] || req.headers['x-api-key'];
+  if (!apiKey) {
+    throw new Error("X-API-Key Header doesn't exist");
+  }
+  return apiKey;
+};
```

Comment:
again, i don't think we should adding new logic this file - rather you should just be moving already existing utils to this file

the reason for this is because this function `fetchAPIKey` will short circuit the logic where a developer is running the project locally but doesn't have an API Key in their header - it will cause the server to return an error which is unexpected behavior

i would be open to having this change in a separate PR but not in this one, please revert this util since it changes the already existing functionality that shouldn't be changed within this PR